client/galaxy/scripts/mvc/workflow/workflow-terminals.js

```diff
@@ -476,11 +476,29 @@ define(['mvc/workflow/workflow-globals'], function( Globals ) {
         initialize: function( attr ) {
             Terminal.prototype.initialize.call( this, attr );
             this.datatypes = attr.datatypes;
-            this.collectionType = new CollectionTypeDescription( attr.collection_type );
-            this.isCollection = true;
+            if( attr.collection_type ) {
+                this.collectionType = new CollectionTypeDescription( attr.collection_type );
+            } else {
+                var collectionTypeSource = attr.collection_type_source;
+                if(! collectionTypeSource) {
```

Comment:
Here and in the update function there is a space missing before the closing parens.